invenio_rdm_records/services/access/service.py

```diff
@@ -0,0 +1,761 @@
+            )
+            if request.is_open
+        ]
+
+        if any((req.is_open for req in requests)):
+            raise DuplicateAccessRequestError([str(r.id) for r in requests])
+
+        record = self.record_cls.pid.resolve(id_)
+        data = {
+            "payload": {
+                "permission": "view",
+                "message": message,
+            }
+        }
+
+        # TODO might also be community managers
+        receiver = None
+        record_owner = record.parent.access.owner.resolve()
+        if record_owner:
+            receiver = record_owner
+
+        if receiver is None:
+            # TODO as fallback maybe admins?
+            pass
+
+        request = current_requests_service.create(
+            identity,
+            data,
+            UserAccessRequest,
+            receiver,
+            topic=record,
+            expires_at=None,  # TODO expiration date
+            expand=expand,
+            uow=uow,
+        )
+
+        # immediately submit the request, unless it has errors
+        # TODO is this correct?
+        if request.errors:
+            return request
+
+        message = {
+            "payload": {
+                "content": data["payload"].get("message") or _("No comment"),
+            }
+        }
+
+        # TODO can we pass the message here?
+        return current_requests_service.execute_action(
+            identity,
+            request.id,
+            "submit",
+            data=message,
+            uow=uow,
+        )
+
+    @unit_of_work()
+    def create_guest_access_request_token(
+        self, identity, id_, data, expand=False, uow=None
+    ):
+        """Create an request token that can be used to create an access request."""
+        # TODO proper permission checks
+        if authenticated_user in identity.provides:
+            raise PermissionDeniedError("request_guest_access")
+
+        record = self.record_cls.pid.resolve(id_)
+        if current_app.config.get("MAIL_SUPPRESS_SEND", False):
+            current_app.logger.warn(
+                "Cannot proceed with guest based access request - "
+                "email sending has been disabled!"
+            )
+
+        access_token = AccessRequestToken.create(
+            email=data["email"],
+            full_name=data["full_name"],
+            message=data["message"],
+            record_pid=id_,
+            shelf_life=timedelta(hours=6),
+        )
+
+        # TODO `url_for()` doesn't work as is because it's creating an /api/ URL...
+        verify_url = url_for(
+            "invenio_rdm_records_ext.verify_access_request_token",
+            _external=True,
+            **{"access_request_token": access_token.token},
+        ).replace("/api/", "/")
+
+        uow.register(
+            EmailOp(
+                receiver=data["email"],
+                subject=_(
+                    "Access request for '%(record_title)s'",
```

Comment:
@rekt-hard is the `subject` automatically prefixing something, e.g. the site name?
```
[ZENODO] bla bla bla
```
What about the footer?
We should make sure that there is coherence in the e-mail content.

Reply:
As of now, there is nothing like that.
It sure would be nice to have a base template for emails. Maybe also a macro for the subject to prefix, as you mentioned.